Hope I got this right...

DavidAnderson684 opened the pull request:
Please pull WordPress.php...

Commit:
WordPress.php Storage helper

Commit:
New WordPress consumer

Commit:
Add callback + resumption capability to chunkUpload

BenExile commented:
Hi @DavidAnderson684 

As you're aware we spoke about some of these features at length and upload resumption has been added to the library. I know you said you were doing some work on the WordPress storage handler too - where are you with this? Is it worth me merging these changes now or not?

DavidAnderson684 commented:
Hi Ben,

The WordPress storage handler seems to work fine, so feel free to merge 
it - it's being used by thousands of people. You can get a copy of the 
code I'm using from: http://wordpress.org/extend/plugins/updraftplus.

I also implemented resuming downloads - since this is just an HTTP Range 
header, it's not very complex, but here it is anyway; just two 
modifications:
1. New parameter on getFile, defaulting to false for backwards 
   compatibility:

public function getFile($file, $outFile = false, $revision = null, 
$allow_resume = false)
1. Tweak the logic for opening the destination file, depending on 
   whether $allow_resume is true and the file exists; and add the Range 
   header as appropriate:
   
   ```
    if ($outFile !== false) {
        // Create a file handle if $outFile is specified
        if ($allow_resume && file_exists($outFile)) {
           if (!$handle = fopen($outFile, 'a')) {
               throw new Exception("Unable to open file handle for 
   ```
   
   $outFile");
               } else {
                   $this->OAuth->setOutFile($handle);
                   $params['headers'] = array('Range: 
   bytes='.filesize($outFile).'-');
               }
            }
            elseif (!$handle = fopen($outFile, 'w')) {
                throw new Exception("Unable to open file handle for 
   $outFile");
            } else {
                $this->OAuth->setOutFile($handle);
            }
        }

If the file is already 100% downloaded, then that should throw an 
exception (you should get an HTTP Range Not Satisfiable response). But 
that doesn't seem necessarily bad. In my use case, I check the file size 
before calling this routine anyway.

David

On 15/03/13 20:39, Ben Tadiar wrote:

> Hi @DavidAnderson684 https://github.com/DavidAnderson684
> 
> As you're aware we spoke about some of these features at length and 
> upload resumption has been added to the library. I know you said you 
> were doing some work on the WordPress storage handler too - where are 
> you with this? Is it worth me merging these changes now or not?
> 
> —
> Reply to this email directly or view it on GitHub 
> https://github.com/BenTheDesigner/Dropbox/pull/66#issuecomment-14983653.

## 

"For the wages of sin is death; but the gift of God is eternal life through Jesus Christ our Lord." - Romans 6:23

Professional and friendly website/e-mail/domain name services: http://www.simbahosting.co.uk

Homepage: http://david.dw-perspective.org.uk
Blog: http://mothwo.blogspot.com

DavidAnderson684 commented:
Actually, I don't think I've changed the WordPress Storage handler since 
I last sent you a copy or pushed into github. The version in the plugin 
has had name-spacing removed for PHP 5.2 compatibility, so you probably 
don't want to bother with that.

David

On 15/03/13 22:58, David Anderson wrote:

> Hi Ben,
> 
> The WordPress storage handler seems to work fine, so feel free to 
> merge it - it's being used by thousands of people. You can get a copy 
> of the code I'm using from: 
> http://wordpress.org/extend/plugins/updraftplus.
> 
> I also implemented resuming downloads - since this is just an HTTP 
> Range header, it's not very complex, but here it is anyway; just two 
> modifications:
> 1. New parameter on getFile, defaulting to false for backwards 
>    compatibility:
> 
> public function getFile($file, $outFile = false, $revision = null, 
> $allow_resume = false)
> 1. Tweak the logic for opening the destination file, depending on 
>    whether $allow_resume is true and the file exists; and add the Range 
>    header as appropriate:
>    
>    ```
>    if ($outFile !== false) {
>        // Create a file handle if $outFile is specified
>        if ($allow_resume && file_exists($outFile)) {
>           if (!$handle = fopen($outFile, 'a')) {
>               throw new Exception("Unable to open file handle for 
>    ```
>    
>    $outFile");
>               } else {
>                   $this->OAuth->setOutFile($handle);
>                   $params['headers'] = array('Range: 
>    bytes='.filesize($outFile).'-');
>               }
>            }
>            elseif (!$handle = fopen($outFile, 'w')) {
>                throw new Exception("Unable to open file handle for 
>    $outFile");
>            } else {
>                $this->OAuth->setOutFile($handle);
>            }
>        }
> 
> If the file is already 100% downloaded, then that should throw an 
> exception (you should get an HTTP Range Not Satisfiable response). But 
> that doesn't seem necessarily bad. In my use case, I check the file 
> size before calling this routine anyway.
> 
> David
> 
> On 15/03/13 20:39, Ben Tadiar wrote:
> 
> > Hi @DavidAnderson684 https://github.com/DavidAnderson684
> > 
> > As you're aware we spoke about some of these features at length and 
> > upload resumption has been added to the library. I know you said you 
> > were doing some work on the WordPress storage handler too - where are 
> > you with this? Is it worth me merging these changes now or not?
> > 
> > —
> > Reply to this email directly or view it on GitHub 
> > https://github.com/BenTheDesigner/Dropbox/pull/66#issuecomment-14983653.
> 
> ## 
> 
> "For the wages of sin is death; but the gift of God is eternal life through Jesus Christ our Lord." - Romans 6:23
> 
> Professional and friendly website/e-mail/domain name services:http://www.simbahosting.co.uk
> 
> Homepage:http://david.dw-perspective.org.uk
> Blog:http://mothwo.blogspot.com

## 

"For the wages of sin is death; but the gift of God is eternal life through Jesus Christ our Lord." - Romans 6:23

Professional and friendly website/e-mail/domain name services: http://www.simbahosting.co.uk

Homepage: http://david.dw-perspective.org.uk
Blog: http://mothwo.blogspot.com
